st2common/st2common/util/param.py

```diff
@@ -170,17 +170,80 @@ def _process_defaults(G, schemas):
                 _process(G, name, value.get("default"))
 
 
+def _check_any_bad(G, nodes, check_any_bad=None, tracked_parents=None):
+    """
+    :param G: nx.DiGraph
+    :param nodes: list[dict]
+    :param check_any_bad: list[boolean]
+    :param tracked_parents: list[str]
+    """
+    if tracked_parents is None:
+        tracked_parents = []
+    if check_any_bad is None:
+        check_any_bad = []
+    for name in nodes:
+        if "value" not in G.nodes[name] and "template" not in G.nodes[name]:
+            # this is a string not a jinja template;  embedded {{sometext}}
+            check_any_bad.append(False)
+        else:
+            check_any_bad.append(True)
+        children = [i for i in G.predecessors(name)]
+        if name not in tracked_parents:
+            tracked_parents.extend(nodes)
+            _check_any_bad(G, children, check_any_bad, tracked_parents)
+    return check_any_bad
+
+
+def _remove_bad(g_copy, parent, nodes, tracked_parents=None):
+    """
+    :param G: nx.DiGraph
+    :param nodes: str
+    :param check_any_bad: list[str]
+    :param tracked_parents: list[str]
+    """
+
+    if tracked_parents is None:
+        tracked_parents = [parent]
+    else:
+        tracked_parents.append(parent)
+    for i in nodes:
+        g_copy.nodes[parent]["value"] = g_copy.nodes[parent].pop("template")
```

Comment:
Is there a case where value would be populated but template would not?

Reply:
yes, for nodes that are not jinja  templates.  They only have a value.

Reply:
I added a check for the "templates" key.  I actually just bumped into this on my own instance running the fork with this fix.